### PR TITLE
DPL Analysis: port bulk IO patch

### PR DESCRIPTION
**Pull request by ktf**
Adapted to compile with the new code layout. Not yet tested.

### CODE REVIEW

**Comment by ktf** [comment]
@jgrosseo @pzhristov with this version, I have all the standard tests in hyperloop working fine, with quite a nice speedup and with reasonable results, as far as I can tell. If you could cross check with my tag (O2Physics::20210820-1) and after it passes all the tests here, I would suggest we merge this ASAP, so that we can spot early potential troubles elsewhere.

**Comment by pzhristov** [comment]
@ktf Yes, I agree we should merge it ASAP.

File: Framework/Core/src/TableTreeHelpers.cxx [hunk 1]
```diff
@@ -443,32 +421,26 @@ TTree* TableToTree::process()
   }
 
 // is used in TreeToTable
-ColumnIterator::ColumnIterator(TTreeReader& reader, const char* colname)
+ColumnIterator::ColumnIterator(TTree* tree, const char* colname)
 {
+  mBranch = tree->GetBranch(colname);
```
Comment (pzhristov) [inline Framework/Core/src/TableTreeHelpers.cxx:426]: Are we sure the pointers are not 0x0?

File: Framework/Core/src/TableTreeHelpers.cxx [hunk 2]
```diff
@@ -443,32 +421,26 @@ TTree* TableToTree::process()
...
-  // find branch
-  auto tree = reader.GetTree();
-  if (!tree) {
-    LOGP(FATAL, "Can not locate tree!");
-    return;
-  }
-
-  auto br = tree->GetBranch(colname);
-  if (!br) {
+  if (!mBranch) {
```
Comment (pzhristov) [inline Framework/Core/src/TableTreeHelpers.cxx:429]: This check comes too late, we already use mBranch.

File: Framework/Core/src/TableTreeHelpers.cxx [hunk 3]
```diff
@@ -678,99 +606,81 @@ void ColumnIterator::reserve(size_t s)
...
+arrow::Status appendValues(Builder builder, TBuffer& buffer, int64_t size)
+{
+  return builder->AppendValues(reinterpret_cast<T const*>(buffer.GetCurrent()), size, nullptr);
+}
+
+size_t ColumnIterator::push()
 {
   arrow::Status stat;
 
+  TBufferFile buffer{TBuffer::EMode::kWrite, 64 * 1024};
```
Comment (pzhristov) [inline Framework/Core/src/TableTreeHelpers.cxx:619]: What is 64 * 1024?

**Comment by jgrosseo** [comment]
@ktf This is great progress. Can we wait with merging until the LHCC meetings are over. These are on Monday and Tuesday afternoon.

File: Framework/Core/src/TableTreeHelpers.cxx (hunk 3 above)
Comment (ktf) [reply Framework/Core/src/TableTreeHelpers.cxx:619]: A round number... :-) we can probably tweak this.

File: Framework/Core/src/TableTreeHelpers.cxx (hunk 2 above)
Comment (ktf) [reply Framework/Core/src/TableTreeHelpers.cxx:429]: I did not want to change the logic too much, but indeed it looks wrong at the moment.

**Pull request merged**